cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java

```diff
@@ -198,6 +198,13 @@ public BuildCommand() {
             "generation")
     private String graalVMBuildOptions;
 
+    @CommandLine.Option(names = "--optimize", description = "generate optimized executable jar", defaultValue = "false")
+    private Boolean optimizeCodegen;
+
+    @CommandLine.Option(names = "--verbose", description = "generate codegen optimization reports",
```

Comment:
Are you suggesting to add a single flag that contains both `--optimize` and `--verbose`? 
Something like `--optimize-verbose`?

So our flags will be,
`--optimize` - for optimized build
`--optimize-verbose` - for optimized build with report emit

Please correct me if I'm wrong.

Reply:
After an offline discussion with @poorna2152 we decided to keep the two flags the same but add a validation. This ensures `--verbose` can only be used alongside `--optimize`. Using `--verbose` without `--optimize` will now trigger an error. Validation was added in a944fe96e880b941b27523335080741de81e9b7e.

Reply:
As @poorna2152 mentioned, `--verbose` is too generic. We must support verbosity in all the tasks that take place in the compilation (e.g. package resolution). Let's use a more specific term like  `--optimize-verbose`

Reply:
Are you suggesting to add `--optimize-report` as a standalone flag? So when we use `bal build --optimize-report` command, it will generate the optimized executable + optimization reports.

Or shall we use something like `--generate-report` along with `--optimize` flag? Then we can use the `bal build --optimize --generate-report` for the same result .

Reply:
My suggestion is also to have 2 flags. The only concern is the name. `--generate-report` is also too generic IMO. Here are some examples of other similar cases:
`bal test --test-report` - this generates an HTML report
`bal build --dump-graph` - this prints the dependency graph to the console
 
Similarly, we can use `--optimize-report` if it is a proper report or `--dump-optimizations` (the wording can be better)  if we are printing it to the console.

Reply:
`--optimize-report` seems like the better name. Changed the flag name in 1a166f6c99b430d241f9a4a7dc602a898006fcff. We will still have validation to stop `--optimize-report` being used without `--optimize` flag.